docs/WebTransport.md

```diff
@@ -0,0 +1,246 @@
+
+Kestrel currently implements most of the WebTransport [draft-02](https://ietf-wg-webtrans.github.io/draft-ietf-webtrans-http3/draft-ietf-webtrans-http3.html) specification, except for datagrams. Datagrams will be implemented at a later date. This document outlines how to use the already implemented functionality.
+
+# Running the sample app
+To help applications get started on implementing WebTransport, there is the `WebTransportSampleApp` project located at `src\Servers\Kestrel\samples\WebTransportSampleApp`. To use it, simply run it from VS. This will launch the server and a terminal which will show logs from Kestrel as it interacts with the client. Now you should be able to connect to the sample from any client that implements the standard WebTransport draft02 specification.
+
+**Note:** Once you run the `WebTransportSampleApp`, it will print the certificate hash that it is using for the SSL connection. You will need to copy it into your client to make sure that both the server and the client use the same one.
+
+## Using Edge or Chrome DevTools as a client
+The Chromium project has implemented a WebTransport client and can be accessed via their JS API from the Chrome or Edge DevTools console. A good sample app demonstrating how to use that API can be found [here](https://github.com/myjimmy/google-webtransport-sample/blob/ee13bde656c4d421d1f2a8e88fd71f572272c163/client.js).
+
+# Note about preview features
+WebTransport is a preview feature. Therefore, you must manually enable it via the `EnablePreviewFeatures` property and toggle the `Microsoft.AspNetCore.Server.Kestrel.Experimental.WebTransportAndH3Datagrams` `AppContextSwitch`. This can be done by either of the following options:
+- Adding the following `ItemGroup` to your csproj file:
+```xml
+  <ItemGroup>
+    <RuntimeHostConfigurationOption Include="Microsoft.AspNetCore.Server.Kestrel.Experimental.WebTransportAndH3Datagrams" Value="true" />
+  </ItemGroup>
+```
+- Adding the following C# statement somewhere in your project's setup logic:
+```C#
+AppContext.SetSwitch("Microsoft.AspNetCore.Server.Kestrel.Experimental.WebTransportAndH3Datagrams", true);
+```
+**Note:** Setting an `AppContextSwitch` is global. Therefore, the setting applies to all areas of your application, not just the current scope.
+
+# Overview of the Kestrel WebTransport API
+## Setting up a connection
+To setup a WebTransport connection, you will first need to configure a host upon which you open a port. A very minimal example is shown below:
+```C#
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var hostBuilder = new HostBuilder()
+            .ConfigureWebHost(webHost =>
+            {
+                webHost.UseKestrel()
+                .ConfigureKestrel((context, options) =>
+                {
+                    options.Listen([SOME IP ADDRESS], [SOME PORT], listenOptions =>
+                    {
+                        listenOptions.UseHttps([YOUR CERTIFICATE]);
```

Comment:
Mention that the cert requirements are stricter and that the dev cert doesn't work. Show your workaround.